tests/test_datasets/test_utils.py

```diff
@@ -514,3 +517,298 @@ def test_timestamp_range_fail_type(start, end, periods, freq):
 def test_timestamp_range_fail_num_parameters(start, end, periods, freq):
     with pytest.raises(ValueError, match="Of the three parameters: .* must be specified"):
         _ = timestamp_range(start=start, end=end, periods=periods, freq=freq)
+
+
+@pytest.fixture
+def df_aligned_datetime() -> pd.DataFrame:
+    df = generate_ar_df(start_time="2020-01-01", periods=10, n_segments=2, freq="D")
+    return df
+
+
+@pytest.fixture
+def df_aligned_int() -> pd.DataFrame:
+    df = generate_ar_df(start_time=10, periods=10, n_segments=2, freq=None)
+    return df
+
+
+@pytest.fixture
+def df_misaligned_datetime() -> pd.DataFrame:
+    df = generate_ar_df(start_time="2020-01-01", periods=10, n_segments=2, freq="D")
+    df = df.iloc[:-3]
+    return df
+
+
+@pytest.fixture
+def df_misaligned_int() -> pd.DataFrame:
+    df = generate_ar_df(start_time=10, periods=10, n_segments=2, freq=None)
+    df = df.iloc[:-3]
+    return df
+
+
+@pytest.fixture
+def df_misaligned_datetime_with_missing_values() -> pd.DataFrame:
```

Comment:
Do we define dataset misaligned if it has `NaNs` at the end?

Reply:
We should work on terminology, it will be done in the very end. 

I think that NaNs at the end means that we could cut them out and make segment shorter to align with others. This is that `infer_alignment` does.

Here I just marked this `df` like misaligned because we are going to align it using non-trivial alignment (which has different timestamps for each segment).